joplin/js/editor.js

```diff
@@ -165,67 +166,58 @@ $(function() {
     state.currentLang = currentLang;
 
     // replace brackets with hidden span tags
-    $('ul[class="objects"]')
-      .find('label:contains(" [")')
-      .each(function() {
-        this.innerHTML = this.innerHTML.replace(
-          '[',
-          " <span style='display:none;'>",
-        );
-        this.innerHTML = this.innerHTML.replace(']', '</span>');
-      });
-
-    // Hide stuff that isn't our language
-    // Top level fields
-
-    // find the language tag
-    // TODO: have better variable names that don't collide
-    function getLanguageTag(elem, selector, tag) {
-      if (elem.querySelectorAll(selector)[0].getElementsByTagName(tag)[0]) {
-        var languageTag = elem
-          .querySelectorAll(selector)[0]
-          .getElementsByTagName(tag)[0].innerHTML;
-        return languageTag;
-      }
-    }
-    // compare to current and hide accordingly
-    function toggleLanguageField(elem, languageTag, currentLang) {
-      if (languageTag != null && languageTag != currentLang) {
-        elem.classList.add('hidden');
-      } else {
-        elem.classList.remove('hidden');
+    function replaceLanguageLabels() {
+      const languageLabels = $('ul[class="objects"]').find(
+        'label:contains(" [")',
+      );
+      if (languageLabels.length) {
+        if (typeof state.languageLabels === 'undefined') {
+          state.languageLabels = languageLabels;
+        } else {
+          for (let label in languageLabels) {
+            state.languageLabels.push(languageLabels[label]);
+          }
+        }
+        languageLabels.each(function() {
+          this.innerHTML = this.innerHTML.replace(
+            '[',
+            " <span style='display:none;'>",
```

Comment:
It might be nice to add a comment here to explain how this works, something like: "since our translation lib adds language strings we don't want added to our labels, let's hide them."